things/api.py

```diff
@@ -0,0 +1,141 @@
+        then include tasks with any status value.
+
+    area : str, bool, or None, optional
+        Any valid uuid of an area. Only include tasks matching that area.
+        If the argument is `False`, only include tasks _without_ an area.
+        If the argument is `True`, only include tasks _with_ an area.
+        If the argument is `None`, then ignore the area value, that is,
+        include tasks both with and without a containing area.
+
+    project : str or None, optional
+        Any valid uuid of a project. Only include tasks matching that project.
+        If the argument is `False`, only include tasks _without_ a project.
+        If the argument is `True`, only include tasks _with_ a project.
+        If the argument is `None`, then ignore the project value, that is,
+        include tasks both with and without a containing project.
+
+    heading : str or None, optional
+        Any valid uuid of a heading. Only include tasks matching that heading.
+        If the argument is `False`, only include tasks _without_ a heading.
+        If the argument is `True`, only include tasks _with_ a heading.
+        If the argument is `None`, then ignore the heading value, that is,
+        include tasks both with and without a containing heading.
+
+    **kwargs : optional
+        Optional keyword arguments passed to ``Database``.
+
+    Returns
+    -------
+    list
+        A list of dicts representing Things tasks.
+
+    Examples
+    --------
+    >>> things.tasks()
+    """
+    database = Database(**kwargs)
+    if type == "task":
+        return database.get_task(
+            status=status, area=area, project=project, heading=heading
+        )
+    elif type == "project":
+        result = []
+        matched_projects = database.get_projects(status=status, area=area)
+        for project in matched_projects:
+            # group tasks by heading
+            project["tasks"] = tasks(
+                status=status, project=project["uuid"], heading=False
+            )
+            project["headings"] = {
+                heading["title"]: tasks(status=status, heading=heading["uuid"])
+                for heading in database.get_headings(
+                    status=status, project=project["uuid"]
+                )
+            }
+            result.append(project)
+        return result
+
+
+def areas(include_tasks=False, status="open", **kwargs):
+    """
+    Read areas into a list of dicts.
+
+    Parameters
+    ----------
+    include_tasks : boolean, default False
+        Include tasks and projects for each area.
+
+    status : {'open', 'done', 'canceled', None}, default 'open'
+        Include only tasks and projects with the specified status.
+        The argument `None` includes all statuses.
+
+    **kwargs : optional
+        Optional keyword arguments passed to ``Database``.
+
+    Returns
+    -------
+    list
+        A list of dicts representing Things areas.
```

Comment:
probably including tags, right? If yes, only the tags names or also their uuids?

Reply:
Ah, great point! I forgot about the tags. Feel free to add.
It seems to me, that since tag names have to be unique, it might suffice to list their names only here. There is always the `things.tags` lookup you can use if you want to know more.